pkg/rancher-desktop/pages/containers/logs/_id.vue

```diff
@@ -0,0 +1,604 @@
+                const viewport = this.terminal.rows;
+                const isAtBottom = buffer.viewportY >= buffer.length - viewport;
+
+                if (isAtBottom) {
+                  this.terminal.scrollToBottom();
+                }
+              }
+            },
+            onError: (error) => {
+              console.error('Stream error:', error);
+              this.handleStreamError(error);
+            },
+            onClose: (code) => {
+              this.streamProcess = null;
+              if (code !== 0 && this.isContainerRunning) {
+                this.handleStreamError(new Error(`Stream closed with code ${code}`));
+              }
+            },
+            splitOutputLines: false,
+          },
+        };
+
+        if (this.hasNamespaceSelected) {
+          streamOptions.namespace = this.hasNamespaceSelected;
+        }
+
+        const streamArgs = ['--follow', '--timestamps', '--tail', '10000', this.containerId];
+
+        this.streamProcess = this.ddClient.docker.cli.exec('logs', streamArgs, streamOptions);
+
+        this.reconnectAttempts = 0;
+
+
+      } catch (error) {
+        console.error('Error starting log stream:', error);
+
+        const errorMessages = {
+          'No such container': 'Container not found. It may have been removed.',
+          'permission denied': 'Permission denied. Check Docker access permissions.',
+          'connection refused': 'Cannot connect to Docker. Is Docker running?'
+        };
+
+        const errorKey = Object.keys(errorMessages).find(key => error.message.includes(key));
+        this.error = errorKey ? errorMessages[errorKey] : (error.message || this.t('containers.logs.fetchError'));
+
+      }
+    },
+    stopStreaming() {
+      if (this.streamProcess) {
+        try {
+          this.streamProcess.close();
+        } catch (error) {
+          console.error('Error stopping log stream:', error);
+        }
+        this.streamProcess = null;
+      }
+    },
+    handleStreamError(error) {
+      if (this.reconnectAttempts < this.maxReconnectAttempts && this.isContainerRunning) {
+        this.reconnectAttempts++;
+        const delay = Math.pow(2, this.reconnectAttempts - 1) * 1000;
+        setTimeout(() => {
+          this.startStreaming();
+        }, delay);
+      } else {
+        this.error = 'Streaming error: ' + error.message + (this.reconnectAttempts >= this.maxReconnectAttempts ? ' (max retries exceeded)' : '');
+      }
+    },
+    startContainerChecking() {
+      this.containerCheckInterval = setInterval(() => {
+        this.getContainerInfo();
+      }, 30000);
+    },
+    stopContainerChecking() {
+      if (this.containerCheckInterval) {
+        clearInterval(this.containerCheckInterval);
+        this.containerCheckInterval = null;
+      }
+    },
+    async initializeTerminal() {
+      this.isLoading = false;
+      await this.$nextTick();
+
+      return new Promise((resolve) => {
```

Comment:
Since you're already in an async function and `resolve()` is only called at the end, we might as well not do that?

```ts
async initializeTerminal() {
  this.isLoading = false;
  await this.$nextTick();
  if (this.$refs.terminalContainer) {
    this.terminal = new Terminal({
      …
    });
    …
    window.addEventListener('resize', this.resizeHandler);
  }
}
```

I assume this code used to want the promise and then things changed over time.